### PR TITLE
Improvements to running a single Flutter test

### PR DESCRIPTION
**Pull request by skybrian**
- Allow running it in the debugger.
- Allow user to run a single test by right-clicking, either in
  source code or on a test in the test tree widget.

### CODE REVIEW

File: src/io/flutter/run/test/TestConfigProducer.java [hunk 1]
```diff
@@ -35,31 +38,113 @@ protected TestConfigProducer() {
...
+      }
+      elt = elt.getParent();
+    }
+    return null; // not found
+  }
+
+  @Nullable
+  private String getCalledFunctionName(@NotNull DartCallExpression call) {
+    if (!(call.getFirstChild() instanceof DartReference)) return null;
+    return call.getFirstChild().getText();
```
Comment (pq) [inline src/io/flutter/run/test/TestConfigProducer.java]: Do we need to consider malformed ASTs?  That is,  might there be a case where `getFirstChild()` returns null?

Comment (pq) [inline src/io/flutter/run/test/TestConfigProducer.java]: (Or maybe we'd never get to this path in that case?)

File: src/io/flutter/run/test/TestConfigProducer.java [hunk 2]
```diff
@@ -35,31 +38,113 @@ protected TestConfigProducer() {
...
+
+  @Nullable
+  private String getStringIfNotTemplate(@NotNull DartStringLiteralExpression expr) {
+    if (!expr.getShortTemplateEntryList().isEmpty() || !expr.getLongTemplateEntryList().isEmpty()) {
+      return null; // is a template
+    }
+    // We expect a quote, string part, quote.
+    if (expr.getFirstChild() == null) return null;
+    final PsiElement second = expr.getFirstChild().getNextSibling();
+    if (second.getNextSibling() != expr.getLastChild()) return null; // not three items
```
Comment (pq) [inline src/io/flutter/run/test/TestConfigProducer.java]: Is this something we could easily get under test?

File: src/io/flutter/run/test/TestConfigProducer.java (hunk 1 above)
Comment (skybrian) [inline src/io/flutter/run/test/TestConfigProducer.java]: In this case, the instanceof check handles  the null check as well.

File: src/io/flutter/run/test/TestConfigProducer.java (hunk 2 above)
Comment (skybrian) [inline src/io/flutter/run/test/TestConfigProducer.java]: I'll take a look (and send another patch).

**Pull request merged**